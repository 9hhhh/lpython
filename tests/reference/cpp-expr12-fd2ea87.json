{
    "basename": "cpp-expr12-fd2ea87",
    "cmd": "lpython --no-color --show-cpp {infile}",
    "infile": "tests/expr12.py",
    "infile_hash": "a28e73367996457cb261785aa6032bf884113aab4a9ec1967cd99bbc",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "cpp-expr12-fd2ea87.stdout",
    "stdout_hash": "7c4c8c60b4b84511616c2d9eec2c9997acf5158f6c0641b4850080b7",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}