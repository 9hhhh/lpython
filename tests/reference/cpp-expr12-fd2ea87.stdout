#include <iostream>
#include <string>
#include <vector>
#include <cassert>
#include <cmath>
#include <Kokkos_Core.hpp>
#include <lfortran_intrinsics.h>

template <typename T>
Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
{
    Kokkos::View<T*> r("r", v.size());
    for (size_t i=0; i < v.size(); i++) {
        r(i) = v[i];
    }
    return r;
}

int check()
{
    int _lpython_return_variable;
    int a;
    a = test(2, 2);
    _lpython_return_variable = a;
    return _lpython_return_variable;
}

int test(int a, int b)
{
    int _lpython_return_variable;
    _lpython_return_variable = std::pow(a, b);
    return _lpython_return_variable;
}

